src/java.base/share/classes/java/lang/ClassLoader.java

```diff
@@ -1606,17 +1606,27 @@ protected URL findResource(String name) {
      * </ol>
      * <p>Note that once a class loader is registered as parallel capable, there
      * is no way to change it back.</p>
+     * <p>
+     * In cases where {@code ClassLoader.registerAsParallelCapable} is called from a context where
+     * there is no caller frame on the stack (e.g. when called directly
+     * from a JNI attached thread), {@code IllegalCallerException} is thrown.
```

Comment:
```suggestion
     * In cases where this method is called from a context where the caller is not a subclass
     * {@code ClassLoader} or there is no caller frame on the stack (e.g. when called directly
     * from a JNI attached thread), {@code IllegalCallerException} is thrown.
```

Should mention the non-class loader caller as well.